Source/ARTFormEncode.m

```diff
@@ -0,0 +1,66 @@
+//
+//  ARTFormEncode.m
+//  Ably
+//
+//  Created by Ricardo Pereira on 27/09/2019.
+//  Copyright © 2019 Ably. All rights reserved.
+//
+//  Apple left a form encoder out for some reason.
+//  Code credit to @mxcl. Based from https://github.com/mxcl/OMGHTTPURLRQ/blob/master/Sources/OMGFormURLEncode.m
+//
+
+#import "ARTFormEncode.h"
+
+static inline NSString *enc(id in, NSString *ignore) {
+    NSMutableCharacterSet *allowedSet = [NSMutableCharacterSet characterSetWithCharactersInString:ignore];
+    [allowedSet formUnionWithCharacterSet:[NSCharacterSet URLQueryAllowedCharacterSet]];
+    [allowedSet removeCharactersInString:@":/?&=;+!@#$()',*"];
+
+    return [[in description] stringByAddingPercentEncodingWithAllowedCharacters:allowedSet];
+}
+
+#define enckey(in) enc(in, @"[]")
+#define encval(in) enc(in, @"")
+
+static NSArray *DoQueryMagic(NSString *key, id value) {
```

Comment:
Can you suggest a more descriptive name?

Reply:
@paddybyers This code has been taken from another project, that's why I decided to leave it has it is in the original:

https://github.com/ably/ably-cocoa/blob/8efef941d36566207bf68a7ddd5f5f8b57781e86/Source/ARTFormEncode.m#L8-L9

Should I still change the name of the internal function?